src/promptflow/promptflow/executor/_tool_resolver.py

```diff
@@ -131,33 +131,39 @@ def _resolve_assistant_tool(self, assistant_definition: AssistantDefinition):
-        # Plans are underway to establish dedicated methods for inner function resolution.
-        # It will replace the current approach of creating node to reuse existing node resolution logic.
-        node, predefined_inputs = self._create_node_for_assistant_tool(tool)
-        resolved_tool = self.resolve_tool_by_node(node, convert_input_types=True)
-        func_name = resolved_tool.definition.function
-        definition = self._generate_tool_definition(func_name, resolved_tool.definition.description, predefined_inputs)
-        if resolved_tool.node.inputs:
-            inputs = {name: value.value for name, value in resolved_tool.node.inputs.items()}
-            func = partial(resolved_tool.callable, **inputs)
-        else:
-            func = resolved_tool.callable
-        return AssistantTool(name=func_name, openai_definition=definition, func=func)
+    def _load_tool_as_function(self, node_name: str, tool_def: dict) -> AssistantTool:
+        # We clone the raw tool definition to avoid modifying the original AssistantDefinition object,
+        # ensuring it is picklable as flow inputs
+        updated_tool_def = copy.deepcopy(tool_def)
+        # load ToolSource as object from json string
+        updated_tool_def["source"] = (
+            ToolSource.deserialize(updated_tool_def["source"]) if "source" in updated_tool_def else None
+        )
 
-    def _create_node_for_assistant_tool(self, tool: dict):
+        # load predefined_inputs as dictionary
         predefined_inputs = {}
-        for input_name, value in tool.get("predefined_inputs", {}).items():
+        for input_name, value in updated_tool_def.get("predefined_inputs", {}).items():
             predefined_inputs[input_name] = InputAssignment.deserialize(value)
-        node = Node(
-            name="assistant_node",
-            tool="assistant_tool",
-            inputs=predefined_inputs,
-            source=ToolSource.deserialize(tool["source"]) if "source" in tool else None,
-            type=ToolType.PYTHON if "tool_type" in tool and tool["tool_type"] == "python" else None,
+
+        # load the module and Tool object
+        m, tool = self._tool_loader.load_tool_for_assistant_node(node_name, updated_tool_def)
+
+        # construct the resolved inputs dictionary
+        updated_inputs = self._get_assistant_function_input_types(node_name, predefined_inputs, tool, m)
+        callable, init_args = BuiltinsManager._load_tool_from_module(
+            m, tool.name, tool.module, tool.class_name, tool.function, updated_inputs
         )
-        return node, list(predefined_inputs.keys())
+        self._remove_init_args(updated_inputs, init_args)
+
+        # construct the AssistantTool object from the updated inputs + Tool object
+        func_name = tool.function
+        definition = self._generate_tool_definition(func_name, tool.description, predefined_inputs)
+        if updated_inputs:
```

Comment:
This is actually `predefined_inputs`? Why we use `updated_inputs` here? It is not that clear.

Reply:
updated